src/components/Questionnaire/QuestionTypes/DiagnosisQuestion.tsx

```diff
@@ -89,6 +86,176 @@ const DIAGNOSIS_INITIAL_VALUE: Omit<DiagnosisRequest, "encounter"> = {
   dirty: true,
 };
 
+function DiagnosisDatePicker({
+  onsetDatetime,
+  onChange,
+  disabled,
+  hasId,
+}: {
+  onsetDatetime?: string;
+  onChange: (date: Date | undefined) => void;
+  disabled?: boolean;
+  hasId: boolean;
+}) {
+  return (
+    <CombinedDatePicker
+      value={onsetDatetime ? new Date(onsetDatetime) : undefined}
+      onChange={onChange}
+      dateFormat="P"
+      disabled={disabled || hasId}
+      buttonClassName="h-8 md:h-9 w-full justify-start font-normal"
+    />
+  );
+}
+
+function ClinicalStatusSelect({
+  status,
+  onValueChange,
+  disabled,
+}: {
+  status: string;
+  onValueChange: (value: string) => void;
```

Comment:
let's use proper types please
```suggestion
  status: DiagnosisClinicalStatus;
  onValueChange: (value: DiagnosisClinicalStatus) => void;
```